rcl/src/rcl/node.c

```diff
@@ -258,79 +256,57 @@ rcl_node_init(
   RCL_CHECK_FOR_NULL_WITH_MSG(
     node->impl->logger_name, "creating logger name failed", goto fail);
 
-  // node rmw_node_handle
-  if (node->impl->options.domain_id == RCL_NODE_OPTIONS_DEFAULT_DOMAIN_ID) {
-    // Find the domain ID set by the environment.
-    get_env_error_str = rcutils_get_env(ROS_DOMAIN_ID_VAR_NAME, &ros_domain_id);
-    if (NULL != get_env_error_str) {
-      RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
-        "Error getting env var '" RCUTILS_STRINGIFY(ROS_DOMAIN_ID_VAR_NAME) "': %s\n",
-        get_env_error_str);
-      goto fail;
-    }
-    if (ros_domain_id) {
-      unsigned long number = strtoul(ros_domain_id, NULL, 0);  // NOLINT(runtime/int)
-      if (number == ULONG_MAX) {
-        RCL_SET_ERROR_MSG("failed to interpret ROS_DOMAIN_ID as integral number");
-        goto fail;
-      }
-      domain_id = (size_t)number;
-    }
-  } else {
-    domain_id = node->impl->options.domain_id;
+  domain_id = node->impl->options.domain_id;
+  if (RMW_RET_OK != rcl_domain_id(&domain_id)) {
```

Comment:
RCL_RET_OK ?

Reply:
Thanks! Corrected in https://github.com/ros2/rcl/pull/515/commits/0317e66b33fd479dddfc38a8297a31d8801957a3.